driver-core/src/main/com/mongodb/operation/ChangeStreamBatchCursor.java

```diff
@@ -0,0 +1,153 @@
+import org.bson.BsonDocument;
+import org.bson.RawBsonDocument;
+
+import java.util.ArrayList;
+import java.util.List;
+
+final class ChangeStreamBatchCursor<T> implements BatchCursor<T> {
+    private final ReadBinding binding;
+    private final ChangeStreamOperation<T> changeStreamOperation;
+
+    private boolean checked = false;
+    private BsonDocument resumeToken;
+    private BatchCursor<RawBsonDocument> wrapped;
+
+    ChangeStreamBatchCursor(final ChangeStreamOperation<T> changeStreamOperation,
+                            final BatchCursor<RawBsonDocument> wrapped,
+                            final ReadBinding binding) {
+        this.changeStreamOperation = changeStreamOperation;
+        this.resumeToken = changeStreamOperation.getResumeToken();
+        this.wrapped = wrapped;
+        this.binding = binding.retain();
+    }
+
+    BatchCursor<RawBsonDocument> getWrapped() {
+        return wrapped;
+    }
+
+    @Override
+    public boolean hasNext() {
+        return resumeableOperation(new Function<BatchCursor<RawBsonDocument>, Boolean>() {
+            @Override
+            public Boolean apply(final BatchCursor<RawBsonDocument> queryBatchCursor) {
+                return queryBatchCursor.hasNext();
+            }
+        });
+    }
+
+    @Override
+    public List<T> next() {
+        return resumeableOperation(new Function<BatchCursor<RawBsonDocument>, List<T>>() {
+            @Override
+            public List<T> apply(final BatchCursor<RawBsonDocument> queryBatchCursor) {
+                return convertResults(queryBatchCursor.next());
+            }
+        });
+    }
+
+    @Override
+    public List<T> tryNext() {
+        return resumeableOperation(new Function<BatchCursor<RawBsonDocument>, List<T>>() {
+            @Override
+            public List<T> apply(final BatchCursor<RawBsonDocument> queryBatchCursor) {
+                return convertResults(queryBatchCursor.tryNext());
+            }
+        });
+    }
+
+    @Override
+    public void close() {
+        wrapped.close();
+        binding.release();
+    }
+
+    @Override
+    public void setBatchSize(final int batchSize) {
+        wrapped.setBatchSize(batchSize);
+    }
+
+    @Override
+    public int getBatchSize() {
+        return wrapped.getBatchSize();
+    }
+
+    @Override
+    public ServerCursor getServerCursor() {
+        return wrapped.getServerCursor();
+    }
+
+    @Override
+    public ServerAddress getServerAddress() {
+        return wrapped.getServerAddress();
+    }
+
+    @Override
+    public void remove() {
+        throw new UnsupportedOperationException("Not implemented yet!");
+    }
+
+    private List<T> convertResults(final List<RawBsonDocument> rawDocuments) {
+        List<T> results = null;
+        if (rawDocuments != null) {
+            results = new ArrayList<T>();
+            for (RawBsonDocument rawDocument : rawDocuments) {
+                if (!checked && !rawDocument.containsKey("_id")) {
```

Comment:
No need for "checked" here (it's never even updated), as it's ok to throw every time.  After the first time they shouldn't be trying again anyway.